aws/Full-PAS-Deployment.yaml

```diff
@@ -0,0 +1,1607 @@
+          files:
+            /etc/awslogs/awslogs.conf:
+              content: !Sub |
+                [general]
+                state_file= /var/awslogs/state/agent-state
+                [/var/log/cloud-init.log]
+                file = /var/log/cloud-init.log
+                log_group_name = ${LogGroup}
+                log_stream_name = {instance_id}/cloud-init.log
+                datetime_format = 
+                [/var/log/cloud-init-output.log]
+                file = /var/log/cloud-init-output.log
+                log_group_name = ${LogGroup}
+                log_stream_name = {instance_id}/cloud-init-output.log
+                datetime_format = 
+                [/var/log/cfn-init.log]
+                file = /var/log/cfn-init.log
+                log_group_name = ${LogGroup}
+                log_stream_name = {instance_id}/cfn-init.log
+                datetime_format = 
+                [/var/log/cfn-wire.log]
+                file = /var/log/cfn-wire.log
+                log_group_name = ${LogGroup}
+                log_stream_name = {instance_id}/cfn-wire.log
+                datetime_format =               
+              mode: '000444'
+              owner: root
+              group: root
+            /etc/awslogs/awscli.conf:
+              content: !Sub |
+                [plugins]
+                cwlogs = cwlogs
+                [default]
+                region = ${AWS::Region}
+              mode: '000444'
+              owner: root
+              group: root
+          commands:
+            01_create_state_directory:
+              command: mkdir -p /var/awslogs/state
+          services:
+            sysvinit:
+              awslogs:
+                enabled: 'true'
+                ensureRunning: 'true'
+                files:
+                  - /etc/awslogs/awslogs.conf
+        install_psmp:
+          commands:
+            01-CreateCredFile:
+              command: !Sub >-
+                sudo /opt/CARKpsmp/bin/createcredfile /root/CD-Image/user.cred
+                Password -Username Administrator -Password ${VaultAdminPassword}
+                -Hostname
+            02-PSMPdeploy:
+              command:
+                'Fn::Sub':
+                  - >-
+                    /root/CD-Image/register_and_activation.sh
+                    /root/CD-Image/user.cred ${VaultIpAddress} $(curl
+                    http://169.254.169.254/latest/meta-data/instance-id) y
+                  - VaultIpAddress: !Sub '${VaultMachine.PrivateIp},${VaultDRMachine.PrivateIp}'
+            03-PostInstall:
+              command: sudo rm -rf /root/CD-Image/
+            99-SignalCompletion:
+              command: !Sub >-
+                /opt/aws/bin/cfn-signal -e $? --stack ${AWS::StackId} --resource
+                PSMPMachine --region ${AWS::Region}
+    DependsOn:
+      - PSMMachine
+    CreationPolicy:
+      ResourceSignal:
+        Timeout: PT10M
+    DeletionPolicy: Retain
+  InstancesSSMPolicy:
```

Comment:
@avishayil why do we have this Policy and also AmazonEC2RoleforSSM on ComponentsInstanceProfile???
we need to examine this issue